spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/JobOperations.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.rest.client;
+
+import org.springframework.cloud.dataflow.rest.resource.JobExecutionResource;
+import org.springframework.cloud.dataflow.rest.resource.JobInstanceResource;
+import org.springframework.cloud.dataflow.rest.resource.StepExecutionProgressInfoResource;
+import org.springframework.cloud.dataflow.rest.resource.StepExecutionResource;
+import org.springframework.hateoas.PagedResources;
+
+/**
+ * Interface defining operations available for jobs.
+ *
+ * @author Glenn Renfro
+ */
+public interface JobOperations {
+
+	/**
+	 * List job executions known to the system.
+	 */
+	PagedResources<JobExecutionResource> executionList();
+
+	/**
+	 * List job instances for the specified jobName.
+	 */
+	PagedResources<JobInstanceResource> instanceList(String jobName);
+
+	/**
+	 * List job executions known to the system filtered by job name.
+	 *
+	 * @param jobName of the executions.
+	 */
+	PagedResources<JobExecutionResource> executionListByJobName(String jobName);
+
+	/**
+	 * Return the {@link JobExecutionResource} for the id specified.
+	 *
+	 * @param id identifier of the job execution
+	 * @return {@link JobExecutionResource}
+	 */
+	JobExecutionResource viewJobExecution(long id);
```

Comment:
These two methods use a different naming convention compared to the others (viewXXX - verb, versus a noun)